controllers/common.go

```diff
@@ -155,7 +155,7 @@ func reconcileDesiredObjects(ctx context.Context, kubeClient client.Client, logg
 			op = result
 			return createOrUpdateErr
 		})
-		if crudErr != nil && errors.Is(crudErr, manifests.ImmutableChangeErr) {
+		if crudErr != nil && errors.As(crudErr, &manifests.ImmutableChangeErr) {
```

Comment:
why is this changing? don't we want to check for immutable errors here?

Reply:
We still check for immutable errors here.
E.g. the [tests/e2e/statefulset-features](https://github.com/open-telemetry/opentelemetry-operator/blob/e84193dec62615635d4533b7696f1cf27f8ec209/tests/e2e/statefulset-features/01-update-volume-claim-templates.yaml#L17) uses this functionality, and it still passes.
If my understanding is correct, the `errors.As` checks whether an error is a certain type, while `errors.Is` is kinda comparing for exact error values.
If you wish, I will try to extend/modify the implemented `ImmutableFieldChangeErr` custom error so we can still use the `errors.Is` for comparing the errors.

Reply:
nope, thanks for explaining :)